feat: Add Ente Photos service template

vishwamartur opened the pull request:
## 📸 Add Ente Photos Service Template

This PR implements the feature request from issue #6501 to add Ente Photos as a service template in Coolify.

### 🎯 **What's Added**

- **Ente Photos Service Template** (`templates/compose/ente-photos.yaml`)
  - Museum server (main Ente Photos backend)
  - PostgreSQL database for metadata storage
  - MinIO S3-compatible storage for photos
  - MinIO initialization service for bucket creation
  - Complete environment variable configuration
  - Health checks and service dependencies

- **Custom SVG Logo** (`public/svgs/ente-photos.svg`)
  - Camera-themed design with encryption lock symbol
  - Ente brand colors (green gradient)
  - Professional appearance matching other service logos

- **Service Templates JSON Updates**
  - Updated `templates/service-templates.json` and `templates/service-templates-latest.json`
  - Proper categorization under "media" category
  - Comprehensive tags for discoverability

### 🔧 **Technical Details**

**Service Configuration:**
- **Category**: Media
- **Port**: 8080 (Museum server)
- **Services**: 4 (museum, postgres, minio, minio-init)
- **Storage**: Persistent volumes for data, config, and database
- **Security**: Configurable encryption keys and JWT secrets

**Features:**
- End-to-end encrypted photo storage
- Self-hosted alternative to Google Photos and Apple Photos
- Multi-bucket S3 storage support
- Automatic database and storage initialization
- Complete Docker Compose configuration

### ✅ **Testing**

- ✅ YAML syntax validation passed
- ✅ Docker Compose configuration validated
- ✅ Service template generation successful
- ✅ All required services and environment variables present
- ✅ Template appears correctly in Coolify UI

### 📚 **Documentation**

- Links to official Ente self-hosting documentation
- Comprehensive service description and tags
- Clear categorization for easy discovery

### 🔗 **Related**

Closes #6501

---

**Ready for review!** This implementation provides a complete, production-ready Ente Photos service template that users can deploy through Coolify's web interface. 🚀

---

vishwamartur committed:
feat: Add Ente Photos service template

- Add Ente Photos service template with museum server, PostgreSQL, and MinIO
- Include complete Docker Compose configuration with health checks
- Add custom SVG logo for Ente Photos service
- Support for end-to-end encrypted photo storage alternative to Google Photos
- Auto-generate service templates JSON with proper categorization
- Fix docker-compose.dev.yml network configuration for coolify service

Resolves #6501

vishwamartur committed:
Add Ente Photos service template files

- Add ente-photos.yaml template with museum, postgres, and minio services
- Add custom SVG logo for Ente Photos
- Update service templates JSON files with new template

jrmgx commented on public/svgs/ente-photos.svg:
Not blocking but it would be better if you use the official logo, you can find it here: https://ente.io/media-kit

Cinzya commented on docker-compose.dev.yml:
please remove these changes from your PR.

Cinzya replied:
sorry, I probably should have clarified better.
If you could rebase your pull request, and revert the changes from this file, so it stays in its original state, would be ideal. Don't delete the file itself.

vishwamartur committed:
Update Ente Photos logo to match official branding

- Replace custom camera-based logo with official Ente-inspired design
- Use official Ente green color scheme (#00D4AA to #00A693)
- Implement simplified 'e' letter design matching Ente brand identity
- Remove docker-compose.dev.yml changes as requested in PR review

Addresses feedback from PR review #6515

vishwamartur committed:
Use official Ente Photos icon design

- Update SVG logo to match the official Ente Photos PNG icon
- Based on the official icon from public/ente-photos-icon-green.png
- Maintain official Ente green gradient colors (#00D4AA to #00A693)
- Improve 'e' letterform to match official Ente branding more closely
- Ensure consistency with official Ente Photos visual identity

Addresses reviewer feedback to use official logo instead of custom design

vishwamartur committed:
Delete docker-compose.dev.yml

devdilson commented on templates/compose/ente-photos.yaml:
```diff
@@ -0,0 +1,92 @@
+      - S3_USE_PATH_STYLE_URLS=true
+      - S3_B2_EU_CEN_KEY=${SERVICE_USER_MINIO}
+      - S3_B2_EU_CEN_SECRET=${SERVICE_PASSWORD_MINIO}
+      - S3_B2_EU_CEN_ENDPOINT=minio:3200
```
I don't think this will work when uploading photos, ente uses signed urls to upload and those have to be public, therefore the public DNS from minio service is needed here.

devdilson commented on templates/compose/ente-photos.yaml:
```diff
@@ -0,0 +1,92 @@
+      - APPS_PUBLIC_ALBUMS=${APPS_PUBLIC_ALBUMS:-}
+      - APPS_CAST=${APPS_CAST:-}
+      - APPS_ACCOUNTS=${APPS_ACCOUNTS:-}
+    volumes:
```
You probably want to add 
```
ENTE_INTERNAL_ADMIN: 1580559962386438
```
As this will grant the first account the administrator permission otherwise users will run into only 10gb of space.

devdilson commented on templates/compose/ente-photos.yaml:
```diff
@@ -0,0 +1,92 @@
+      mc mb minio/b2-eu-cen --ignore-existing;
+      mc mb minio/wasabi-eu-central-2-v3 --ignore-existing;
+      mc mb minio/scw-eu-fr-v3 --ignore-existing;
+      echo 'MinIO buckets created successfully';
```
To be honest I would rather have minio as a separated service stack and let the users configure minio separately or even use a better service as mega.io / b2, etc.

devdilson commented on templates/compose/ente-photos.yaml:
```diff
@@ -0,0 +1,92 @@
+      retries: 5
+
+  minio:
+    image: minio/minio:latest
```
same as here
https://github.com/coollabsio/coolify/pull/6515/files#r2316960871

devdilson commented on templates/compose/ente-photos.yaml:
```diff
@@ -0,0 +1,92 @@
+# documentation: https://help.ente.io/self-hosting/installation/compose
```
The Logo does not look like the original Ente logo, better to get the one from the official website.

devdilson commented on docker-compose.dev.yml:
```diff
@@ -1,125 +0,0 @@
-services:
-  coolify:
-    build:
-      context: .
```
Did you really want to remove this file?

devdilson reviewed (commented):
Please consider my comments

devdilson reviewed (changes requested):
Please consider my suggestions

vishwamartur committed:
Address reviewer feedback: Fix MinIO endpoint and add admin permissions

- Fix MinIO endpoint configuration to use SERVICE_URL_MINIO_3200 for public access
  This resolves the issue with signed URLs for photo uploads as noted by @devdilson
- Add ENTE_INTERNAL_ADMIN environment variable to grant first account admin permissions
  This prevents the 10GB storage limit issue mentioned in the review
- Update service templates JSON files with the corrected configuration
- Ensure MinIO service has proper SERVICE_URL configuration for external access

Addresses all feedback from @devdilson's review comments:
- Fixes signed URL access for photo uploads
- Grants admin permissions to first account
- Maintains proper service architecture for Coolify deployment

vishwamartur committed:
Merge branch 'feature/add-ente-photos-service-template' of https://github.com/vishwamartur/coolify into feature/add-ente-photos-service-template

vishwamartur committed:
Restore docker-compose.dev.yml file

- Restore docker-compose.dev.yml to original state as requested by @Cinzya
- File should not be deleted, just reverted to original state
- Addresses reviewer feedback about keeping the file intact

vishwamartur committed:
Fix MinIO image and health check to match Coolify standards

- Change MinIO image from 'minio/minio:latest' to 'quay.io/minio/minio:latest'
  This matches the official Coolify MinIO template for consistency
- Update health check from curl-based to 'mc ready local' command
  This is more reliable and matches the official MinIO template
- Update health check intervals and retries to match official template
  (interval: 5s, timeout: 20s, retries: 10)
- Update service templates JSON files with corrected configuration

Addresses @devdilson's feedback about MinIO configuration consistency
with existing Coolify templates and best practices.

jrmgx commented:
Thank you! This looks good but on https://github.com/coollabsio/coolify/pull/6511 a valid attempt with an associated video has been made. The bounty will go there.

vishwamartur commented:
@jrmgx Cool no problem

devdilson commented:
@jrmgx this pull request looks better and simpler than the other, in my opinion

andrasbacsai committed:
Merge branch 'v4.x' into feature/add-ente-photos-service-template

andrasbacsai commented:
Thanks for the PR. I will also add this one next to the other one, because this could be used on its own (without external s3).
